.tools/weathertop/backend/src/main/java/com/weathertop/lambda/HandleRunFargateTask.java

```diff
@@ -0,0 +1,111 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.weathertop.lambda;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.LambdaLogger;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.weathertop.service.FargateTaskRunner;
+import java.util.Map;
+
+public class HandleRunFargateTask implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+    @Override
+    public APIGatewayProxyResponseEvent handleRequest(APIGatewayProxyRequestEvent event, Context context) {
+        LambdaLogger logger = context.getLogger();
+
+        try {
+            // Extract "language" from query parameters
+            Map<String, String> queryParams = event.getQueryStringParameters();
+            if (queryParams == null || !queryParams.containsKey("language")) {
+                return createErrorResponse(400, "Missing required query parameter: language");
+            }
+
+            String language = queryParams.get("language").toLowerCase(); // Normalize
+            logger.log("Received language: " + language);
+
+            FargateTaskRunner runner = new FargateTaskRunner();
+            String taskRunId = "";
+            if (language.compareTo("java") == 0) {
+                String clusteerName = "MyJavaWeathertopCluster";
+                String defName = "WeathertopJava";
+                taskRunId = runner.runFargateTask(defName, clusteerName);
+
+            } else if (language.compareTo("rustv1") == 0) {
+                String clusteerName = "MyRustWeathertopCluster";
```

Comment:
The runner line is repeated 10 times, for maintainability move it out of the switch statement.